mlem/contrib/tensorflow.py

```diff
@@ -0,0 +1,216 @@
+import os.path
+import tempfile
+from typing import Any, ClassVar, Iterator, Optional, Tuple
+
+import h5py
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.keras.saving.saved_model_experimental import sequential
+
+from mlem.constants import PREDICT_METHOD_NAME
+from mlem.core.artifacts import Artifacts, Storage
+from mlem.core.data_type import (
+    DataHook,
+    DataReader,
+    DataSerializer,
+    DataType,
+    DataWriter,
+)
+from mlem.core.errors import DeserializationError, SerializationError
+from mlem.core.hooks import IsInstanceHookMixin
+from mlem.core.model import (
+    BufferModelIO,
+    ModelHook,
+    ModelIO,
+    ModelType,
+    Signature,
+)
+from mlem.core.requirements import InstallableRequirement, Requirements
+
+
+class TFTensorDataType(
+    DataType, DataSerializer, DataHook, IsInstanceHookMixin
+):
+    """
+    :class:`.DataType` implementation for `tensorflow.Tensor` objects
+    which converts them to built-in Python lists and vice versa.
+
+    :param shape: shape of `tensorflow.Tensor` objects in data
+    :param dtype: data type of `tensorflow.Tensor` objects in data
+    """
+
+    type: ClassVar[str] = "tensorflow"
+    valid_types: ClassVar = (tf.Tensor,)
+    shape: Tuple[Optional[int], ...]
+    dtype: str
+
+    def _check_shape(self, tensor, exc_type):
+        if tuple(tensor.shape)[1:] != self.shape[1:]:
+            raise exc_type(
+                f"given tensor is of shape: {(None,) + tuple(tensor.shape)[1:]}, expected: {self.shape}"
+            )
+
+    def serialize(self, instance: tf.Tensor):
+        self.check_type(instance, tf.Tensor, SerializationError)
+        if instance.dtype is not getattr(tf, self.dtype):
```

Comment:
add proprety for `getattr(tf, self.dtype)`?

Reply:
done